### ISSUE TITLE
Duplicate ids passed into langchain\vectorstores\fais.__add will result in a mismatch between faiss and document's index

### ISSUE THREAD

**Issue by xia-huang-411303** [langchain-ai/langchain#9019]
### System Info

windows 10 
python 3.10.12

### Who can help?

@eyurtsev  There is a BUG in the def __add function in langchain\vectorstores\faiss.py: When the parameter ids is provided and there are duplicates in it, index_to_docstore_id will report an error and the program will terminate. But the text vector embedding has been added to the index. This causes the id in the index to be inconsistent with the id in index_to_docstore_id, not aligned, and the text cannot be matched correctly.

langchain\vectorstores\faiss.py 中 def __add 函数存在BUG:
当参数 ids 已提供, 且其中存在重复时, index_to_docstore_id 新增会报错,程序运行终止. 但是文本向量 embedding 已经添加至 index 中.导致 index 中的 id 与 index_to_docstore_id 的中 id 对应不一致,文本无法正确匹配

### Information

- [x] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [x] Document Loaders
- [x] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

None

### Expected behavior

None

**Comment by xia-huang-411303** [comment]
CN: 这个解决方案不太行.因为相重复的ID可以是反复调用_add方法时传入的,单次调用的检查去重无法避免这个问题.可能需要在向FAISS中加入向量时,检查ID是否已经存在于index_to_docstore_id中,或者只是改变代码的顺序,如下. 总而言之,FAISS的ID不能跟文档的ID保持强一致性是一个硬伤

EN: This solution is not very good. Since duplicate ids can be passed in multiple calls to _add, single-call deduplication cannot avoid this problem. You may need to check if the ID already exists in index_to_docstore_id when adding a vector to FAISS, or just change the order of the code, as follows:

```python
    def __add(
        self,
        texts: Iterable[str],
        embeddings: Iterable[List[float]],
        metadatas: Optional[List[dict]] = None,
        ids: Optional[List[str]] = None,
        **kwargs: Any,
    ) -> List[str]:
        if not isinstance(self.docstore, AddableMixin):
            raise ValueError(
                "If trying to add texts, the underlying docstore should support "
                f"adding items, which {self.docstore} does not"
            )
        documents = []
        for i, text in enumerate(texts):
            metadata = metadatas[i] if metadatas else {}
            documents.append(Document(page_content=text, metadata=metadata))
        if ids is None:
            ids = [str(uuid.uuid4()) for _ in texts]
        # Add to the index, the index_to_id mapping, and the docstore.
        starting_len = len(self.index_to_docstore_id)
        faiss = dependable_faiss_import()
        # Get list of index, id, and docs.
        full_info = [(starting_len + i, ids[i], doc) for i, doc in enumerate(documents)]
        # Add information to docstore and index.
        self.index_to_docstore_id.update({index: _id for index, _id, _ in full_info})
        self.docstore.add({_id: doc for _, _id, doc in full_info})
        vector = np.array(embeddings, dtype=np.float32)
        if self._normalize_L2:
            faiss.normalize_L2(vector)
        self.index.add(vector)
        return [_id for _, _id, _ in full_info]
```
In summary, the fact that FAISS ids are not strongly aligned with the document ids is a pain
